tests/TestUtils.hs

```diff
@@ -209,10 +211,23 @@ compile SupportModules{..} inputFiles = runTest $ do
   tell $ foldMap (\(fp, (ws, _)) -> CST.toMultipleWarnings fp ws) msWithWarnings
   let ms = fmap snd <$> msWithWarnings
   foreigns <- inferForeignModules ms
-  let actions = makeActions supportModules (foreigns `M.union` supportForeigns)
+  let
+    actions = makeActions supportModules (foreigns `M.union` supportForeigns)
+    hasMainModule = (==) 1 $ length $ filter (== T.pack "Main") $ fmap getPsModuleName ms
   case ms of
-    [singleModule] -> pure <$> P.rebuildModule actions supportExterns (snd singleModule)
-    _ -> P.make actions (CST.pureResult <$> supportModules ++ map snd ms)
+    [singleModule] -> do
+      unless hasMainModule $ do
+        error $ "When testing a single PureScript file, the file's module's name must be 'Main' but got '"
+          <> T.unpack (getPsModuleName singleModule) <> "'."
+      pure <$> P.rebuildModule actions supportExterns (snd singleModule)
+    _ -> do
+      unless hasMainModule $ do
+        error "When testing a multiple PureScript files, the main file's module's name must be 'Main'."
```

Comment:
I'm guessing `error` isn't what we want here, but this at least got the ball rolling.

Reply:
Seems fine to me. But I would drop:

```suggestion
        error "When testing multiple PureScript files, the main file's module's name must be 'Main'."
```